instrumentation/http/src/main/java/brave/http/HttpResponseParser.java

```diff
@@ -0,0 +1,153 @@
+
+  /**
+   * Implement to choose what data from the http response are parsed into the span representing it.
+   *
+   * <p>Note: Either the response or error parameters may be null, but not both.
+   *
+   * @see Default
+   */
+  void parse(@Nullable HttpResponse response, @Nullable Throwable error, TraceContext context,
+    SpanCustomizer customizer);
+
+  /**
+   * The default data policy sets the span name to the HTTP route when available, and sets the and
+   * adds the "http.status_code" and "error" tags.
+   *
+   * <p><h3>Route-based span name</h3>
+   * If routing is supported, and a GET didn't match due to 404, the span name will be "get
+   * not_found". If it didn't match due to redirect, the span name will be "get redirected". If
+   * routing is not supported, the span name is left alone.
+   */
+  // This accepts response or exception because sometimes http 500 is an exception and sometimes not
+  // If this were not an abstraction, we'd use separate hooks for response and error.
+  class Default implements HttpResponseParser {
+
+    /**
+     * This tags "http.status_code" when it is not 2xx. If the there is no exception and the status
+     * code is neither 2xx nor 3xx, it tags "error". This also overrides the span name based on the
+     * {@link HttpResponse#method()} and {@link HttpResponse#route()} where possible (ex "get
+     * /users/:userId").
+     *
+     * <p>If you only want to change how exceptions are parsed, override {@link #error(int,
+     * Throwable, TraceContext, SpanCustomizer)} instead.
+     */
+    @Override public void parse(@Nullable HttpResponse response, @Nullable Throwable error,
+      TraceContext context, SpanCustomizer customizer) {
+      int statusCode = 0;
+      if (response != null) {
+        statusCode = response.statusCode();
+        String nameFromRoute = spanNameFromRoute(response, statusCode);
+        if (nameFromRoute != null) customizer.name(nameFromRoute);
+        String maybeStatus = maybeStatusAsString(statusCode, 299);
+        if (maybeStatus != null) customizer.tag("http.status_code", maybeStatus);
+      }
+      error(statusCode, error, context, customizer);
+    }
+
+    /** The intent of this is to by default add "http.status_code", when not a success code */
+    @Nullable String maybeStatusAsString(int statusCode, int upperRange) {
+      if (statusCode != 0 && (statusCode < 200 || statusCode > upperRange)) {
+        return String.valueOf(statusCode);
+      }
+      return null;
+    }
+
+    static String spanNameFromRoute(HttpResponse response, int statusCode) {
+      String method = response.method();
+      if (method == null) return null; // don't undo a valid name elsewhere
+      String route = response.route();
+      if (route == null) return null; // don't undo a valid name elsewhere
+      if (!"".equals(route)) return method + " " + route;
+      if (statusCode / 100 == 3) return method + " redirected";
```

Comment:
I guess most 3xx are redirects, but not 304 Not Modified.

Reply:
agreed. fixed